patches/minecraft/net/minecraft/util/IChatComponent.java.patch

```diff
@@ -8,3 +8,44 @@
      String func_150254_d();
  
      List<IChatComponent> func_150253_a();
+@@ -57,7 +56,7 @@
+ 
+                         for (JsonElement jsonelement : jsonarray1)
+                         {
+-                            IChatComponent ichatcomponent2 = this.deserialize(jsonelement, jsonelement.getClass(), p_deserialize_3_);
++                            IChatComponent ichatcomponent2 = p_deserialize_3_.deserialize(jsonelement, IChatComponent.class);
+ 
+                             if (ichatcomponent1 == null)
+                             {
+@@ -91,12 +90,13 @@
+ 
+                         if (jsonobject.has("with"))
+                         {
++                            if (!jsonobject.isJsonArray()) throw new JsonParseException("'with' tag has to be an array");
```

Comment:
Are you sure this check is correct? It seems as if this should be `if (!jsonarray.isJsonArray())` below the next line?